pypuf/learner/evolution_strategies/becker.py

```diff
@@ -0,0 +1,311 @@
+        self.individuals = np.zeros((parent_size, instance.n))  # weight_arrays
+        # mean, step_size,  pop_size,   parent_size,    priorities, cov_matrix,   path_cm,    path_ss
+        # m,    sigma,      lambda,     mu,             w_i,        C,            p_c,        p_sigma
+        self.mean = np.zeros(instance.n) # mean vector of distribution
+        self.step_size = 1              # distance to next distribution
+        self.pop_size = pop_size        # number of individuals per generation
+        self.parent_size = parent_size  # number of considered individuals
+        self.priorities = priorities    # array of consideration proportion
+        self.cov_matrix = np.identity(instance.n)  # shape of distribution
+        self.path_cm = np.matrix(np.zeros(instance.n)) # cumulated evolution path of covariance matrix
+        self.path_ss = np.zeros(instance.n) # cumulated evolution path of step size
+        # auxiliary constants
+        self.mu_w = 1 / np.sum(np.square(priorities))
+        self.c_mu = self.mu_w / instance.n**2
+        self.d_sigma = 1 + np.sqrt(self.mu_w / instance.n)
+        self.c_1 = 2 / instance.n**2
+        self.c_c = 4 / instance.n
+        self.c_sigma = 4 / instance.n
+        assert len(priorities) == parent_size
+        assert int(np.ndarray.sum(priorities)) == 1
+        assert (self.c_1 + self.c_mu <= 1)
+
+    def learn(self):
+        # this is the main learning method
+        # returns XOR-LTFArray with nearly same behavior as learned instance
+        self.epsilon = self.learn_epsilon(self.instance, self.unreliability)
+        while self.num_of_LTFs < self.instance.k:
+            self.new_LTF = self.learn_LTF()
+            if self.is_different_LTF():
+                self.different_LTFs[self.num_of_LTFs] = self.new_LTF
+                self.num_of_LTFs += 1
+        self.different_LTFs = self.set_pole_of_LTFs(self.instance, self.different_LTFs)
+        return LTFArray(self.different_LTFs, self.transform, self.combiner, bias=False)
+
+    def is_different_LTF(self):
+        # returns True iff new_LTF is different from previously learned LTFs
+        if self.num_of_LTFs == 0 :
+            return True
+        weight_arrays = self.different_LTFs[:self.num_of_LTFs-1, :]
+        new_LTFArray = LTFArray(self.new_LTF, self.transform, self.combiner)
+        different_LTFs = self.build_LTFArrays(weight_arrays)
+        challenges = tools.sample_inputs(self.instance.n, self.challenge_num)
+        responses = np.empty((self.num_of_LTFs, self.challenge_num))
+        for i, challenge in enumerate(challenges):
+            responses[0, i] = new_LTFArray.eval(challenge)
+            for j, current_LTF in enumerate(different_LTFs):
+                responses[j+1, i] = current_LTF.eval(challenge)
+        return not self.is_correlated(responses)
+
+    def learn_LTF(self):
+        # this is the main CMA-ES algorithm like that from Hansen
```

Comment:
Please use [Python doc comments](https://www.python.org/dev/peps/pep-0257/#specification) instead of just `#` whenever you comments on what a function or class does.